### DISCUSSION TITLE
Is there an option for openid-connect plugin passing all requests to an upstream if no token provided?

### DISCUSSION THREAD

**Discussion by purefun** [apache/apisix#5349] [Q&A] [answered]
I ran into a scenario that an upstream should receive all requests:

1. ✅ `bearer_only` should set true, the client should not redirected to openid server login endpoint
2. ❓ If access token is not provided, the plugin passes the request to the upstream
4. ❓ If access token is invalid, send the request to the upstream still
5. ✅ if access token is valid, set `X-userinfo` in header, then passes request to the upstreamHow are you yes

Is there an option for openid-connect plugin passing all requests to an upstream if no token provided? Than you.

**Comment by tokers** [comment]
@starsz PTAL.

**Comment by purefun** [comment, marked as answer]
It's impossible to pass requests to the upstream if token is not provided.

https://github.com/apache/apisix/blob/042ce5c1a6a4e040d6d6924f2d86e5534b730134/apisix/plugins/openid-connect.lua#L170-L173

**Reply by starsz** [reply]
Yes. But what's the scenario that `If access token is not provided, the plugin passes the request to the upstream` ?

**Reply by purefun** [reply]
Some of the interfaces contained in an upstream require login and some do not.
